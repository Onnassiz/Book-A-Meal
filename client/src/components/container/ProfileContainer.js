import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { postProfile, getProfile, updateProfile, putImage } from '../../reducer/profile/actionsDispatcher';
import Profile from '../presentation/Profile';

export const mapStateToProps = state => ({
  user: state.user,
  profile: state.profile,
  formState: state.formState,
});


export function mapDispatchToProps(dispatch) {
  return bindActionCreators({
    updateProfile,
    postProfile,
    putImage,
    getProfile,
  }, dispatch);
}


export default withRouter(connect(
  mapStateToProps,
  mapDispatchToProps,
)(Profile));
