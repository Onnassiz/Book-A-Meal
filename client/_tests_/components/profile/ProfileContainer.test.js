import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';
import ProfileContainer, { mapStateToProps, mapDispatchToProps } from '../../../src/components/container/ProfileContainer';

const mockStore = configureStore();

const store = mockStore({
  profile: {},
  user: {},
  formState: { isLoading: false },
});

const initialState = {
  profile: {},
  user: {},
  formState: { isLoading: false },
};

const wrapper = shallow(<ProfileContainer store={store} />);

describe('<ProfileContainer />', () => {
  it('Should render the Profile container successfully', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).getProfile();
    expect(mapStateToProps(initialState).profile).toEqual({});
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
