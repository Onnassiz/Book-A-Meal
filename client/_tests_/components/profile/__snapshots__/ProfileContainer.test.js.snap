// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileContainer /> Should render the Profile container successfully 1`] = `
<Route
  render={[Function]}
/>
`;
